webui/react/src/hooks/useSettings.ts

```diff
@@ -280,6 +297,34 @@ const useSettings = <T>(config: SettingsConfig, options?: SettingsHookOptions):
     updateSettings(newSettings);
   }, [ config.settings, updateSettings ]);
 
+  const fetchUserSetting = useCallback(async () => {
+
+    if(!user) return;
+    const userSettingResponse = await getUserSetting({ userId: user.id });
+    userSettingResponse.settings.forEach(setting => {
+      const { key, value, storagePath } = setting;
+      const jsonValue = JSON.parse(value || '');
+      const config = configMap[key];
+      if (!config) return;
+      const isValid = validateSetting(config, jsonValue);
+      const isDefault = isEqual(config.defaultValue, jsonValue);
+
+      // Store or clear setting if `storageKey` is available.
+      if (config.storageKey && isValid) {
+        if (jsonValue === undefined || isDefault) {
+          storage.remove(config.storageKey, storagePath);
+        } else {
+          storage.set(config.storageKey, jsonValue, storagePath);
+        }
+      }
+    });
+
+  }, [ configMap, storage, user ]);
+
+  useEffect(() => {
+    fetchUserSetting();
```

Comment:
would this cause `settings` to be updated, if `settings` was defined by `const { settings } = useSettings(...)`?